### PR TITLE
docs: add gcp project admin guide and reference

### PR DESCRIPTION
**Pull request by jgehrcke**

### CODE REVIEW

File: docs/guides/administrator/gcp-setup.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
+
+# How to set up a GCP project for Opstrace
+
+This is a detailed walk-through showing how to set up a fresh GCP project for Opstrace.
+
+## Step 1: create a project
+
+Start by creating a fresh project with the [GCP UI](https://console.cloud.google.com/projectcreate).
+In addition to using a good project name (which you can change later), be sure to choose an expressive **project ID** on that screen (as the UI says: **it cannot be changed later**).
```
Comment (jgehrcke) [inline docs/guides/administrator/gcp-setup.md:9]: Note to self: we recommend to pick the ID first, and then set the name to the ID to prevent confusion?

File: docs/guides/administrator/gcp-setup.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+Log in using the same GCP account that you created the project with above.
+
+
+Review `gcloud`'s authentication state:
+
+```bash
+gcloud auth list
+```
+
+Are you sure that `gcloud` is now logged in with and using the GCP account that you created the project with above?
```
Comment (jgehrcke) [inline docs/guides/administrator/gcp-setup.md:33]: note to self: typo, reword

File: docs/guides/administrator/gcp-setup.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+```bash
+gcloud auth list
+```
+
+The remaining steps assume that `gcloud` is now in fact logged in to a GCP account that has admin privileges for the newly created project.
+
+In the current terminal, set these two environment variables:
+
+```text
```
Comment (triclambert) [inline docs/guides/administrator/gcp-setup.md:38]: A drive-by glance...  wondering why you switched to `text` from `bash`?

Comment (jgehrcke) [reply docs/guides/administrator/gcp-setup.md:38]: In my experience across publishing platforms, `text` usually yields the more predictable syntax highlighting results when showing individual commands. `bash` syntax highlighting can sometimes be rather colorful/wrong for individual commands :) That's why I often prefer `text` when don't see the result yet. Depends on the syntax highlighting solution of course. Will have to get used to our solution here :).

Comment (triclambert) [reply docs/guides/administrator/gcp-setup.md:38]: Understood.  Let's try to dial this in so we can have proper highlighting in our docs eventually.

File: docs/references/gcp.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+* `roles/compute.networkAdmin`
+* `roles/container.admin`
+* `roles/editor`
+* `roles/iam.securityAdmin`
+
+If you would like to know which individual permissions are implied by any of these roles, please search for that role (e.g. `roles/container.admin`) on the [GCP IAM permissions reference page](https://cloud.google.com/iam/docs/permissions-reference).
+
+Note: we didn't quite arrive at fulfilling the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) here -- this is an ongoing effort.
+If you have specific ideas for reducing the set if privileges required, please [open an issue](https://github.com/opstrace/opstrace/issues)!
```
Comment (triclambert) [inline docs/references/gcp.md:31]:
```suggestion
If you have specific ideas for reducing the set of privileges required, please [open an issue](https://github.com/opstrace/opstrace/issues/new/choose)!
```

Let's link them to the issue types page.  That way they can open a bug or even choose to go to our Discussions section.

File: docs/references/gcp.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+That service account must have certain security roles applied in the GCP project:
+
+* `roles/compute.networkAdmin`
+* `roles/container.admin`
+* `roles/editor`
+* `roles/iam.securityAdmin`
+
+If you would like to know which individual permissions are implied by any of these roles, please search for that role (e.g. `roles/container.admin`) on the [GCP IAM permissions reference page](https://cloud.google.com/iam/docs/permissions-reference).
+
+Note: we didn't quite arrive at fulfilling the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) here -- this is an ongoing effort.
```
Comment (triclambert) [inline docs/references/gcp.md:30]:
```suggestion
Note: we didn't quite arrive at fulfilling the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege) here—this is an ongoing effort.
```

If you're ok with it, can we use an actual em dash (U+2014)?

File: docs/references/gcp.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Creating an Opstrace cluster in a GCP project requires the following APIs to be [enabled](https://cloud.google.com/service-usage/docs/enable-disable) in that project:
+
+* Cloud DNS API
+* Compute Engine API
+* Kubernetes Engine API
+* Cloud SQL Admin API
+* Service Networking API
+* Cloud Resource Manager API
+
+We try to keep this list up to date -- if in doubt, please contact us!
```
Comment (triclambert) [inline docs/references/gcp.md:16]:
```suggestion
We try to keep this list up to date—if in doubt, please contact us!
```

File: docs/references/gcp.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# GCP project configuration
+
+Note: For a detailed walk-through showing how to set up a fresh GCP project for Opstrace, please try our corresponding [admin guide](<fillme>).
```
Comment (triclambert) [inline docs/references/gcp.md:3]:
```suggestion
Note: For a detailed walk-through showing how to set up a fresh GCP project for Opstrace, please try our corresponding [admin guide](../guides/administrator/gcp-setup.md).
```

**Comment by triclambert** [comment]
This is great!  I will test the setup guide today.

File: docs/references/gcp.md (hunk 5 above)
Comment (jgehrcke) [reply docs/references/gcp.md:30]: For sure, I am used to ` -- ` being processed by an actual typesetting system into a proper em dash -- which does not happen in GH-rendered MD. So: thanks for noticing :).

File: docs/guides/administrator/gcp-setup.md (hunk 1 above)
Comment (jgehrcke) [reply docs/guides/administrator/gcp-setup.md]: done

File: docs/references/gcp.md (hunk 5 above)
Comment (triclambert) [reply docs/references/gcp.md]: I think we have some Markdownlint rules to build in our future.  ;)

File: docs/guides/administrator/gcp-setup.md [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+chmod 600 "${GCP_PROJECT_ID}-svc-acc.json"
+```
+
+Now, move this credential file wherever you need it to be and remember to always treat it securely: for a production workflow, you may want to submit it to a system designed for storing secrets, and then wipe it from your local machine.
+
+## Further steps
+
+For creating an Opstrace cluster using this project and service account, all you need to do is to set the environment variable `GOOGLE_APPLICATION_CREDENTIALS` (to point to the credentials file created in step 7) before invoking `opstrace create gcp ...`. For further information, please have a look at the [CLI reference documentation](./cli.md).
```
Comment (triclambert) [inline docs/guides/administrator/gcp-setup.md:160]:
```suggestion
For creating an Opstrace cluster using this project and service account, all you need to do is to set the environment variable `GOOGLE_APPLICATION_CREDENTIALS` (to point to the credentials file created in step 7) before invoking `opstrace create gcp ...`. For further information, please have a look at the [CLI reference documentation](../../references/cli.md).
```

I think this is the right path from here.

File: docs/references/aws.md [hunk 9]
```diff
@@ -1,8 +1,6 @@
-# Cloud permissions
+# AWS account configuration
 
-## Required cloud permissions for cluster creation
-
-### AWS
+## Required permissions
 
 Generally speaking, we subscribe to the [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege).
```
Comment (triclambert) [inline docs/references/aws.md:5]: Btw ❤️ the shoutout here.

File: docs/references/gcp.md (hunk 4 above)
Comment (jgehrcke) [reply docs/references/gcp.md]: done: pushed commit

File: docs/references/gcp.md (hunk 5 above)
Comment (jgehrcke) [reply docs/references/gcp.md]: done: pushed commit

File: docs/references/gcp.md (hunk 6 above)
Comment (jgehrcke) [reply docs/references/gcp.md]: done: pushed commit

File: docs/references/gcp.md (hunk 7 above)
Comment (jgehrcke) [reply docs/references/gcp.md]: done: pushed commit

File: docs/references/aws.md (hunk 9 above)
Comment (jgehrcke) [reply docs/references/aws.md:5]: copied from you! :)

File: docs/guides/administrator/gcp-setup.md (hunk 8 above)
Comment (jgehrcke) [reply docs/guides/administrator/gcp-setup.md:160]: Oh right!

Comment (jgehrcke) [reply docs/guides/administrator/gcp-setup.md]: Commit pushed.
Also added a line break between sentences.

File: docs/references/gcp.md (hunk 4 above)
Comment (jgehrcke) [reply docs/references/gcp.md]: update: the if/of is now also fixed on the branch.

**Comment by jgehrcke** [comment]
@triclambert I'm leaving into the weekend I think. Please add more review comments and/or merge at your discretion!

**Review by triclambert — APPROVED** [review]
Lgtm.  Tested successfully!

**Comment by jgehrcke** [comment]
Thanks for doing this over the weekend, Chris!
